src/arithmetic/arithmetic_expression.c

```diff
@@ -585,6 +585,14 @@ AR_Func AR_GetFunc(char *func_name) {
     return NULL;
 }
 
+bool AR_FuncExists(const char *func_name) {
+    char lower_func_name[32] = {0};
+    size_t lower_func_name_len = 32;
+    _toLower(func_name, &lower_func_name[0], &lower_func_name_len);
```

Comment:
`_toLower` will fail an assertion if `func_name` is greater than 32 bytes, but the argument may be up to 1023.